vm/src/builtins/type.rs

```diff
@@ -30,10 +33,56 @@ pub struct PyType {
     pub subclasses: PyRwLock<Vec<PyRef<PyWeak>>>,
     pub attributes: PyRwLock<PyAttributes>,
     pub slots: PyTypeSlots,
+    pub heaptype_ext: Option<Pin<Box<HeapTypeExt>>>,
+}
+
+#[derive(Default)]
+pub struct HeapTypeExt {
+    pub number_methods: PyNumberMethods,
+}
+
+pub struct PointerSlot<T>(NonNull<T>);
```

Comment:
What's the benefit of this type comparing to `&'static`?
They seem to guarantee same things.